sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -122,6 +122,37 @@ object JdbcUtils extends Logging {
     }
   }
 
+  /**
+   * Runs a custom query against a table from the JDBC database.
+   */
+  def runQuery(conn: Connection, actions: String, options: JDBCOptions): Unit = {
+    val autoCommit = conn.getAutoCommit
+    conn.setAutoCommit(false)
+    val queries = actions.split(";")
```

Comment:
@moomindani We probably need to parse it smartly if we want to allow multi statements. Above may not work very well if the queries have embedded semi colons in a string literal ? Also, not sure if some databases allow semi colons in the identifier like table names.

Reply:
How about reusing the existing code? https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala#L521-L578

Reply:
Thank you, it makes sense to me.
I modified it to use the linked code.